src/mimeparser.rs

```diff
@@ -369,6 +370,11 @@ impl MimeMessage {
         parser.heuristically_parse_ndn(context).await;
         parser.parse_headers(context).await?;
 
+        if !decryption_info.dkim_results.allow_keychange {
+            for part in parser.parts.iter_mut() {
+                part.error = Some("Seems like DKIM failed, this either is an attack or (more likely) a bug in Authentication-Results-checking. Please tell us about this at https://support.delta.chat.\n\nScan the sender's QR code to make this work again.".to_string());
```

Comment:
TODO for me: Make sure that the UIs actually show this error (before this PR, we never put errors on incoming messages)

Reply:
I opened a PR at https://github.com/deltachat/deltachat-android/pull/2404